src/components/atoms/TextField/index.tsx

```diff
@@ -0,0 +1,194 @@
+  switch (textFieldType) {
+    case 'email':
+      return 110;
+    case 'password':
+      return 80;
+    case 'confirmPassword':
+      return 100;
+    case 'bookName':
+      return 50;
+    case 'bookPrice':
+      return 40;
+    case 'purchaseDate':
+      return 0;
+  }
+};
+
+const isPassword = (textFieldType: TextFieldType) => {
+  return textFieldType == 'password' || textFieldType == 'confirmPassword';
+};
+
+export function TextFieldComponent(props: Props) {
+  const classes = useStyles();
+  const { textFieldType, password, setText } = props;
+  const [values, setValues] = useState<TextFieldState>({
+    textValue: "",
+    showIcon: false,
+    showPassword: isPassword(textFieldType),
+  });
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleClickShowPassword = () => {
+    setValues({ ...values, showPassword: !values.showPassword });
+  };
+
+  const handleChange = (prop: keyof TextFieldState) => (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    console.log(values.textValue)
+    console.log(event.target.value);
+    if(setText) {
+      setText(event.target.value);
+    }
+    setValues({ ...values, [prop]: event.target.value });
+  };
+
+  const handleMouseDownPassword = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
+
+  const renderPasswordIcon = () => {
+    return (
+      <InputAdornment position="end">
+        <IconButton
+          aria-label="toggle password visibility"
+          onClick={handleClickShowPassword}
+          onMouseDown={handleMouseDownPassword}
+          edge="end"
+        >
+          {values.showPassword ? <VisilityIcon /> : <VisilityIconOff />}
+        </IconButton>
+      </InputAdornment>
+    );
+  };
+
+  const isValid = () => {
+    if (
+      textFieldType == 'password' ||
+      textFieldType == 'email'
+    ) {
+      return !inputRef.current?.validity.valid;
+    }
+
+    if (
+      textFieldType == 'confirmPassword' &&
+      password !== values.textValue
+    ) {
+      return true
+    }
+
+    return false;
+  };
+
+  const textFieldClass = classNames(classes.marginTextField, classes.textField);
+  const helperTextClass = classNames(classes.marginHelperText, classes.helperText);
+
+  return (
+    <TextFieldContainer>
+      <FormControll className={textFieldClass} variant="outlined">
+        <InputLabel htmlFor="outlined">
+          {returnInputLabel(textFieldType)}
+        </InputLabel>
+        <OutlinedInput
+          error={isValid()}
+          id="outlined"
+          type={
+            values.showPassword
+              ? 'password'
+              : returnOutlinedtextFieldType(textFieldType)
+          }
+          onChange={handleChange('textValue')}
+          endAdornment={
+            textFieldType == 'password' || textFieldType == 'confirmPassword'
+              ? renderPasswordIcon()
+              : undefined
```

Comment:
これでどうだろう 🤔 
renderPasswordIcon()をコンポーネント化した場合は呼び出し方が少し変わってきます

```suggestion
            isPassword(textFieldType) && renderPasswordIcon()
```

Reply:
上の記法でうまくいきました！